src/components/TodoList/TodoList.tsx

```diff
@@ -1,224 +1,115 @@
-                <span className="icon">
-                  <i className="far fa-eye" />
-                </span>
-              </button>
-            </td>
-          </tr>
-
-          <tr data-cy="todo">
-            <td className="is-vcentered">8</td>
-
-            <td className="is-vcentered">
-              <span className="icon" data-cy="iconCompleted">
-                <i className="fas fa-check" />
-              </span>
-            </td>
-
-            <td className="is-vcentered is-expanded">
-              <p className="has-text-success">quo adipisci enim quam ut ab</p>
-            </td>
-
-            <td className="has-text-right is-vcentered">
-              <button data-cy="selectButton" className="button" type="button">
-                <span className="icon">
-                  <i className="far fa-eye" />
-                </span>
-              </button>
-            </td>
-          </tr>
-
-          <tr data-cy="todo">
-            <td className="is-vcentered">9</td>
-            <td className="is-vcentered"> </td>
-
-            <td className="is-vcentered is-expanded">
-              <p className="has-text-danger">molestiae perspiciatis ipsa</p>
-            </td>
-
-            <td className="has-text-right is-vcentered">
-              <button data-cy="selectButton" className="button" type="button">
-                <span className="icon">
-                  <i className="far fa-eye" />
-                </span>
-              </button>
-            </td>
-          </tr>
-
-          <tr data-cy="todo">
-            <td className="is-vcentered">10</td>
-
-            <td className="is-vcentered">
-              <span className="icon" data-cy="iconCompleted">
-                <i className="fas fa-check" />
-              </span>
-            </td>
-
-            <td className="is-vcentered is-expanded">
-              <p className="has-text-success">
-                illo est ratione doloremque quia maiores aut
-              </p>
-            </td>
+export const TodoList: React.FC = () => {
+  const [modalOpen, setModalOpen] = useState(false);
+  const [activeTodoId, setActiveTodoId] = useState<number | null>(null);
+  const todos = useAppSelector((state: RootState) => state.todos.items);
+  const filterState = useAppSelector((state: RootState) => state.filter);
+  const { query, status } = filterState;
+  const dispatch = useDispatch();
+
+  const handleTodoClick = (todo: Todo) => {
+    dispatch(setCurrentTodo(todo));
+    setActiveTodoId(todo.id);
+    setModalOpen(true);
+  };
+
+  const filteredTodos = todos.filter((todo: Todo) => {
+    const matchesQuery = todo.title.toLowerCase().includes(query.toLowerCase());
+
+    if (status === 'all') {
+      return matchesQuery;
+    } else if (status === 'active') {
+      return matchesQuery && !todo.completed;
+    } else if (status === 'completed') {
+      return matchesQuery && todo.completed;
+    }
+
+    return false;
+  });
```

Comment:
The `filteredTodos` computation is done inside the component render, which means it will be re-calculated every time the component re-renders. This could be a potential performance issue if the `todos` list is large. Consider using a memoized selector with `reselect` library to compute the filtered todos.